orcestra/postprocess/level1.py

```diff
@@ -148,6 +149,45 @@ def _filter_spikes(ds, threshold=5, window=1200):
     return xr.where(abs(diff) < threshold, ds, interpolated)
 
 
+def _filter_land(ds, mask, lat, lon, offset=pd.Timedelta("7s")):
+    """
+    Filters out data that was collected over land.
+    Removes data by offset earlier than the the time the plane flies over land
+    to avoid including land measurements due to tilt of the plane or mask inaccuracies.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset to filter.
+    mask : xr.DataArray
+        Mask of land and sea. 1 for sea, 0 for land.
+    lat : xr.DataArray
+        Latitudes of the path.
+    lon : xr.DataArray
+        Longitudes of the path.
+    offset : pd.Timedelta
+        Time offset to remove data before the plane flies over land. Default is 7 seconds.
+
+    Returns
+    -------
+    xr.Dataset
+        Filtered dataset.
+    """
+
+    mask_path = mask.sel(lat=lat, lon=lon, method="nearest")
+    mask_path_subsampled = mask_path.sel(time=ds.time, method="nearest").assign_coords(
+        time=ds.time
+    )
+    diff = (mask_path_subsampled * 1).diff("time")
+    start_land = diff.where(diff == -1).dropna("time").time
+    end_land = diff.where(diff == 1).dropna("time").time
+    for t in start_land:
+        mask_path_subsampled.loc[dict(time=slice(t - offset, t))] = 0
+    for t in end_land:
+        mask_path_subsampled.loc[dict(time=slice(t, t + offset))] = 0
```

Comment:
Would it also be possible to just filter the, e.g. 30s running mean of the sea-land track?